src/overlays/actors/ovl_Obj_Kinoko/z_obj_kinoko.c

```diff
@@ -28,12 +28,59 @@ const ActorInit Obj_Kinoko_InitVars = {
     (ActorFunc)ObjKinoko_Draw,
 };
 
-#endif
+void ObjKinoko_Init(Actor* thisx, GlobalContext* globalCtx) {
+    thisx->world.pos.y += 4.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kinoko/ObjKinoko_Init.s")
+void ObjKinoko_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kinoko/ObjKinoko_Destroy.s")
+void ObjKinoko_Update(Actor* thisx, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kinoko/ObjKinoko_Update.s")
+    if (player->currentMask != PLAYER_MASK_SCENTS) {
+        thisx->draw = NULL;
+        thisx->hintId = 0xFF;
+        thisx->flags &= ~ACTOR_FLAG_1;
+    } else {
+        thisx->draw = ObjKinoko_Draw;
+        thisx->hintId = 0x64;
+        thisx->flags |= ACTOR_FLAG_1;
+        if (Actor_HasParent(thisx, globalCtx)) {
+            Flags_SetCollectible(globalCtx, OBJ_KINOKO_PARAMS(thisx));
+            Actor_MarkForDeath(thisx);
+            return;
+        }
+        Actor_PickUp(thisx, globalCtx, GI_MAX, 20.0f, 10.0f);
+        if (Math_SmoothStepToF(&thisx->speedXZ, 0.0f, 0.04f, 2.0f, 0.5f) < 0.5f) {
+            thisx->scale.x = 0.0f;
+            thisx->speedXZ = 110.0f;
+            thisx->velocity.x = 0.2f;
+        }
+        if (Math_SmoothStepToF(&thisx->scale.x, thisx->velocity.x, 0.04f, 0.004f, 0.001f) <
+            ((thisx->velocity.x == 0.0f) ? 0.0f : 0.05f)) {
+            thisx->velocity.x = 0.0f;
+        }
+        thisx->scale.y = thisx->scale.x;
+        thisx->scale.z = thisx->scale.x;
+        thisx->shape.rot.y = func_800DFCDC(globalCtx->cameraPtrs[globalCtx->activeCamera]) + 0x8000;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kinoko/ObjKinoko_Draw.s")
+void ObjKinoko_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    Gfx* gfx;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    func_8012C2DC(globalCtx->state.gfxCtx);
+    gfx = POLY_XLU_DISP;
+    gDPSetPrimColor(&gfx[0], 0, 0, 169, 63, 186, (u8)thisx->speedXZ);
+    gDPSetEnvColor(&gfx[1], 110, 44, 200, 100);
+    gDPSetRenderMode(&gfx[2], G_RM_PASS, G_RM_ZB_CLD_SURF2);
+    gSPMatrix(&gfx[3], Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(&gfx[4], &gMushroomGlow[2]);
```

Comment:
Given that this is not using the whole displaylist, I am somewhat concerned that this is not its only use, and this name might not be sufficiently general. Could you add a note to the xml?

Reply:
That is what prevented me from naming more in gameplay_keep and I even hesitated to name the DL, but I can definitely add a comment to the xml!